woocommerce/woocommerce-android
# Order creation products multi-selection

### Order creation - products multi-selection

Closes: #8420


### Description
- Integrated existing `ProductSelectorFragment` into Order creation - product/variation multi-selection flow. The old solution allowed the selection of a single product. The new one provides a multi-selection of items.
- Deleted old product and variation picker code 
- Modified data model returned by `ProductSelectorViewModel` to distinguish product and variation ids. The previous data model was a single set of product and variation ids mixed together. While this was sufficient for Coupons API for Order creation we need to pass variation and product ids separately.
- Deleted UI tests responsible for testing old product pickers code.

### Testing instructions



Go to the Orders screen and create a new Order. Tap `+ Add product` screen and verify the multi-selection adds products/variations correctly to the order. Important scenarios to test:
- Can select multiple products and variations at the same time
- Products and variations already added to the order are marked as selected when returning to the product picker page
- Unselecting previously selected products/variations should remove them from the order
- In case the selected product quantity was updated on the order page it should be preserved when opening the product picker and returning back to the order
- Tapping on the selected product on the Orders page should redirect to the Item Details screen and allow removing the product

The coupon edition screen (product pickers) should remain working without any changes as well.

### Images/gif

https://user-images.githubusercontent.com/4527432/221597902-ecd6da7b-15bf-44b0-a848-af2c3a7e60f2.mp4
